### PR TITLE
[#2872] Documentation + improved UI

### LINKED ISSUE #2872: Adjust mail to behandelaar
We willen de:

- mail naar behandelaar
- onderwerpregel

kunnen aanpassen zodat we bijvoorbeeld component/gebruikersvariabelen kunnen toevoegen.

From Sivlia:

Dit heeft te maken met de email registratie backend en niet de Object registratie backend in combinatie met een email notificatie. 

DH zou graag de template / subject van de email die gestuurd wordt door de email registratie backend (zoals bij de bevestigings email)

From Sergei:

Before sending registration email, generate registration reference and include it in the context and reuse it during reference extraction (extract public reference task)

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #2872 (last PR! :open_mouth:)

### CODE REVIEW

File: src/openforms/js/components/admin/form_design/registrations/index.js [hunk 1]
```diff
@@ -10,4 +12,15 @@ export const BACKEND_OPTIONS_FORMS = {
     form: CamundaOptionsForm,
     onStepEdit: onStepEdit,
   },
+  email: {
+    uiSchema: {
+      emailContentTemplateText: {
+        'ui:widget': 'textarea',
+        'ui:options': {
+          rows: 5,
```
Comment (SilviaAmAm) [inline src/openforms/js/components/admin/form_design/registrations/index.js:20]: I would have liked to overwrite the `cols` too, but it doesn't look like there is that option: https://github.com/rjsf-team/react-jsonschema-form/blob/main/packages/core/src/components/widgets/TextareaWidget.tsx#L47

If we really want it, i will make a custom widget also for the textarea

Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/registrations/index.js:20]: can't you do this with CSS and give it a particular width?

File: docs/manual/templates.rst [hunk 2]
```diff
@@ -319,3 +324,90 @@ Voorbeeld
...
+
+==================================  ===========================================================================
+Variabele                           Beschrijving
+==================================  ===========================================================================
+``{{ form_name }}``                 De naam van het formulier.
+``{{ completed_on }}``              De tijdstip waarop het formulier werd ingezonden.
+``{{ public_reference }}``          De openbare referentie van de inzending.
+``{{ payment_received }}``          Indicatie of de gebruiker wel of niet heeft betaald.
+``{{ payment_order_id }}``          De referentie van de betaling.
+``{{ submission_language }}``       De taal van het formulier die werd ingezonden.
```
Comment (sergei-maertens) [inline docs/manual/templates.rst:350]: These are the language codes, or not? Could benefit from an example:

```suggestion
``{{ submission_language }}``       De taal van het formulier die werd ingezonden, bijvoorbeeld 'nl'.
```

File: docs/manual/templates.rst [hunk 3]
```diff
@@ -319,3 +324,90 @@ Voorbeeld
...
+``{{ co_signer }}``                 De voorletters en de achternaam van de persoon die het formulier heeft mede-ondertekend.
```
Comment (sergei-maertens) [inline docs/manual/templates.rst:351]: Doesn't/should this now also include the identifying attribute (BSN) with Paul's latest improvements in #2716

Comment (SilviaAmAm) [reply docs/manual/templates.rst:351]: Yes!

**Pull request merged**